### PR TITLE
Garbage cleanup error does not force kubeturbo restarts

### PR DESCRIPTION
**Pull request by irfanurrehman**
Fixes https://vmturbo.atlassian.net/browse/OM-67739

### CODE REVIEW

**Review by chlam4 — APPROVED** [review]

**Review by marvelyue — APPROVED** [review]
LGTM

File: pkg/discovery/worker/garbage_collector.go [hunk 1]
```diff
@@ -163,29 +176,27 @@ func (g *GarbageCollector) isLeakedPod(pod api.Pod) bool {
...
-	}
-
-	if err := g.unpauseRollouts(); err != nil {
-		return err
-	}
-
-	return nil
+func (g *GarbageCollector) revertControllers() {
+	g.revertSchedulers()
+	g.unpauseRollouts()
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply pkg/discovery/worker/garbage_collector.go:181]: They are actually dependent. The revert scheduler puts the pods parent (eg a replicaset or replication controller) in a good state, until which we should ideally not enable the grand parents (eg a deployment) rollout.

**Comment by irfanurrehman** [comment]
Thanks @chlam4 and @marvelyue for checking this.
@ading1977 please merge this if after your due diligence.

File: pkg/discovery/worker/garbage_collector.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chlam4) [reply pkg/discovery/worker/garbage_collector.go:181]: If a revert fails, should we continue the rollout or not to continue?  Before this latest change, the rollout will not happen if the corresponding revert fails.  After this change, the rollout will be called even if the corresponding revert fails.

[the remark this answers is not in the document]
Comment (irfanurrehman) [reply pkg/discovery/worker/garbage_collector.go:181]: As an additional note, we also concluded to put out an improvement at some point later, where we can do this cleanup in parallel to kubeturbo run by making this exclusive to action executions by using some locks, such that the cleanup happens at a time when there is no action in action.

**Pull request merged**